packages/lesswrong/components/ea-forum/GlobalButtonBurst.tsx

```diff
@@ -0,0 +1,183 @@
+      @keyframes fly-out {
+        0% {
+          transform: translate(0px, 0px) rotate(0deg) scale(1);
+          opacity: 1;
+        }
+        100% {
+          transform: var(--transform);
+          opacity: 0;
+        }
+      }
+    `;
+    document.head.appendChild(style);
+
+    onIdle(() => {
+      const attachedButtons = new WeakMap<HTMLButtonElement, EventListener>();
+
+      function handleClick(ev: MouseEvent) {
+        const btn = ev.currentTarget as HTMLButtonElement | null;
+        if (!btn) return;
+
+        const centerX = ev.clientX;
+        const centerY = ev.clientY;
+
+        const newItems: BurstItem[] = Array.from({ length: PNG_COUNT }).map(() => {
+          const id = Math.random().toString(36).substring(2, 11);
+          const angle = randomBetween(0, 2 * Math.PI);
+          const distance = randomBetween(80, 180);
+
+          const endX = centerX + (distance * Math.cos(angle));
+          const endY = centerY + (distance * Math.sin(angle));
+
+          return {
+            id,
+            startX: centerX,
+            startY: centerY,
+            endX,
+            endY,
+            rotate: randomBetween(0, 360),
+            scale: randomBetween(0.5, 1.2),
+          };
+        });
+
+        setBursts((prev) => [...prev, ...newItems]);
+
+        setTimeout(() => {
+          setBursts((prev) =>
+            prev.filter((b) => !newItems.some((ni) => ni.id === b.id))
+          );
+        }, 1000);
+      }
+
+      function attachListener(btn: HTMLButtonElement) {
+        if (!attachedButtons.has(btn)) {
+          const fn: EventListener = (evt) => handleClick(evt as MouseEvent);
+          btn.addEventListener("click", fn, { passive: true });
+          attachedButtons.set(btn, fn);
+        }
+      }
+
+      function detachListener(btn: HTMLButtonElement) {
+        const fn = attachedButtons.get(btn);
+        if (fn) {
+          btn.removeEventListener("click", fn);
+          attachedButtons.delete(btn);
+        }
+      }
+
+      function attachAllButtonsIn(root: ParentNode) {
+        const allBtns = root.querySelectorAll("button");
+        allBtns.forEach((btn) => attachListener(btn as HTMLButtonElement));
+      }
+
+      const observer = new MutationObserver((mutations) => {
+        for (const m of mutations) {
+          if (m.type === "childList") {
+            m.addedNodes.forEach((node) => {
+              if (node instanceof HTMLElement) {
+                if (node.tagName.toLowerCase() === "button") {
+                  attachListener(node as HTMLButtonElement);
+                }
+                attachAllButtonsIn(node);
+              }
+            });
+            m.removedNodes.forEach((node) => {
+              if (node instanceof HTMLElement) {
+                if (node.tagName.toLowerCase() === "button") {
+                  detachListener(node as HTMLButtonElement);
+                }
+                const removedBtns = node.querySelectorAll("button");
```

Comment:
This won't catch every element that has a button-like role, but does catch the majority of them